src/core/net/dnssd_server.cpp

```diff
@@ -1004,6 +1004,8 @@ Error Server::ResolveByUpstream(const Request &aRequest)
     txn = AllocateUpstreamQueryTransaction(*aRequest.mMessageInfo);
     VerifyOrExit(txn != nullptr, error = kErrorNoBufs);
 
+    VerifyOrExit(otPlatDnsIsUpstreamQueryAvailable(&GetInstance()), error = kErrorNoRoute);
```

Comment:
IMHO [kErrorInvalidState](https://github.com/openthread/openthread/blob/293b41c8081b7a7a75114b0f494219e4368980a0/src/core/common/error.hpp#L67) may be more proper than `kErrorNoRoute`. 

cc @abtink What do you think?

Reply:
Yes. Thanks. I fully agree. `InvalidState` seems to be better fit.